bay/images/docker/settings.php

```diff
@@ -101,21 +102,42 @@
       $redis->auth($redis_password);
     }
 
-    $response = $redis->ping();
-    if (strpos($response, 'PONG') === 'FALSE') {
+    // Check the redis mode - to determine which interface we use to connect.
+    $info = $redis->info();
+    $redis_interface = 'PhpRedis';
+
+    if (isset($info['redis_mode']) && $info['redis_mode'] == 'cluster' && class_exists('\Drupal\redis\Cache\PhpRedisCluster')) {
+      $redis_interface = 'PhpRedisCluster';
+    }
+
+    if (getenv('REDIS_INTERFACE')) {
+      // Allow environment variable override for the interface (eg. forcing
+      // connection via a standalone pod).
+      $redis_interface = getenv('REDIS_INTERFACE');
+    }
+
+    if (strpos($redis->ping(), 'PONG') === 'FALSE') {
       throw new \Exception('Redis reachable but is not responding correctly.');
     }
+
     $settings['redis.connection']['host'] = $redis_host;
     $settings['redis.connection']['port'] = $redis_port;
-    $settings['redis.connection']['password'] = '';
+    $settings['redis.connection']['password'] = $redis_password;
     $settings['redis.connection']['base'] = 0;
-    $settings['redis.connection']['interface'] = 'PhpRedis';
+    $settings['redis.connection']['interface'] = $redis_interface;
+    $settings['redis.connection']['read_timeout'] = $redis_timeout;
+    $settings['redis.connection']['timeout'] = $redis_timeout;
     $settings['cache']['default'] = 'cache.backend.redis';
     $settings['cache']['bins']['bootstrap'] = 'cache.backend.chainedfast';
     $settings['cache']['bins']['discovery'] = 'cache.backend.chainedfast';
     $settings['cache']['bins']['config'] = 'cache.backend.chainedfast';
     $settings['container_yamls'][] = $contrib_path . '/redis/example.services.yml';
     $settings['cache_prefix']['default'] = getenv('REDIS_CACHE_PREFIX') ?: getenv('LAGOON_PROJECT') . '_' . getenv('LAGOON_GIT_SAFE_BRANCH');
+
+    if ($redis_interface == 'PhpRedisCluster') {
+      $settings['redis.connection']['seeds'] = ["$redis_host:$redis_port"];
```

Comment:
Unsure if this is needed - the docs are vague on seeds and RedisCluster can gather this information on the connection.

```
$cluster = new RedisCluster(NULL, ['redis-cluster:7000'], 1.5, 1.5, false);
=> RedisCluster {#13406
     _masters: [
       [
         "192.168.16.4",
         7000,
       ],
       [
         "192.168.16.4",
         7002,
       ],
       [
         "192.168.16.4",
         7001,
       ],
     ],
```
Has no visible difference to defining all available hosts in the seed (as per docs).
```
$cluster = new RedisCluster(NULL, ['redis-cluster:7000', 'redis-cluster:7001', 'redis-cluster:7002'], 1.5, 1.5, false);
=> RedisCluster {#13408
     _masters: [
       [
         "192.168.16.4",
         7000,
       ],
       [
         "192.168.16.4",
         7002,
       ],
       [
         "192.168.16.4",
         7001,
       ],
     ],
```

Reply:
The seed seems unimportant as well, if we omit this the module will use `$settings['redis_host'] . ':' . $settings['redis_port'];` as a default (which is essentially what we're doing here).